sema4ai/src/sema4ai_code/vendored_deps/oauth2_settings.py

```diff
@@ -115,6 +115,14 @@ class OAuth2ProviderSettingsResolved(OAuth2ProviderSettings):
         tokenEndpoint="/oauth2/v2.0/token",
         # server=settings.server,
     ),
+    "salesforce": OAuth2ProviderSettings(
+        server="https://login.salesforce.com",
+        tokenEndpoint="/services/oauth2/token",
+        revocationEndpoint="/services/oauth2/revoke",
+        authorizationEndpoint="https://login.salesforce.com/services/oauth2/authorize",
+        authParams={},
+        requiresHttps=True,
```

Comment:
Does Salesforce always force HTTPS or is it optional?
Asking because in localhost use this means user needs to create the self-signed cert etc..

Reply:
Yes, based on their documentation: https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_endpoints.htm&type=5

Reply:
Tested locally with a http callback and it works, will put it as false